### PR TITLE
bump go to 1.17

### PR DESCRIPTION
**Pull request by STRRL**
### What problem does this PR solve?

It's a part of https://github.com/chaos-mesh/chaos-mesh/issues/2752, preparing for upgrading controller-runtime and kubernetes relates dependencies.

### Related changes

- [ ] Need to update `chaos-mesh/website`
- [ ] Need to update `Dashboard UI`
- Need to **cheery-pick to release branches**
  - [x] release-2.1
  - [ ] release-2.0

### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] No code
- [x] Manual test (add steps below)

Side effects

- [ ] Breaking backward compatibility

### Release note 

```text
Please add a release note.

You can safely ignore this section if you don't think this PR needs a release note.
```

### DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

**Comment by STRRL** [comment]
PTAL 

/cc @cwen0 @WangXiangUSTC @YangKeao @Hexilee @iguoyr @g1eny0ung

File: pkg/ptrace/ptrace_nocgo.go [hunk 1]
```diff
@@ -12,6 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 //
+//go:build !cgo
```
Comment (cwen0) [inline pkg/ptrace/ptrace_nocgo.go:15]: This is duplicated with `+build !cgo`?

Comment (STRRL) [reply pkg/ptrace/ptrace_nocgo.go:15]: Yes, I will replace them.

Comment (STRRL) [reply pkg/ptrace/ptrace_nocgo.go:15]: Updated, except `// +build !ignore_autogenerated` in `zz_generated.deepcopy.go`, which is controlled by controller-gen.

**Comment by STRRL** [comment]
Updated PTAL @cwen0 @YangKeao

**Review by Andrewmatilde — APPROVED** [review]
LGTM

**Comment by STRRL** [comment]
> cherry pick to release-2.1 failed

I would cherry-pick to release-2.1 manually.

**Pull request merged**